vowpalwabbit/red_python.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+#include <vector>
+#include "reductions_fwd.h"
+
+VW::LEARNER::base_learner* red_python_setup(VW::config::options_i& options, vw& all);
+
+namespace RED_PYTHON
+{
+    //this should live elsewhere?
+    class ExternalBinding
```

Comment:
this class might as well be named everything_but_base since it holds everything learner needs to init except base learner<ExternalBinding, example>::init_learner

Reply:
this will all be simplified as we develop the custom stack building, which will make it easier to do sane operations on the chain of learners.